### DISCUSSION TITLE
OAuth using keycloak get claims from user info endpoint

### DISCUSSION THREAD

**Discussion by dagolav** [supabase/supabase#17574] [Questions]
Hi! 
I am testing out supabase-js, supabase.auth.signInWithOAuth, provider is keycloak. 
It works to sign in, but how do I get claims from user info endpoint? I need to get additional claims like roles and organization number for using with row level security. 
Is it possible? 

Kind regards 
Dag Olav

**Comment by olee** [comment]
Hi, I have the same issue.
Did anyone else or you @dagolav find a solution for this?

**Reply by bhargavi-17** [reply]
@olee did you got the answer

**Reply by olee** [reply]
We did find some solution by getting the provider token through supabase auth and then using this through keycloak API to fetch the user attributes iirc.
Cannot give a definite answer right now though as I wasn't the one who implemented it in our codebase.
